### PR TITLE
Update filesystem and fsstat docs to update the definition of a "file"

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## What does this PR do?

I recently realized that what the `system/fsstat` and `system/filesystem` calls `files` are in fact _inodes_. A lot of users might not even notice the difference, as the total count of files and used inodes can be fairly similar, but I figured it was at least worth updating the docs for anything that decides to dig deeper. It would be nice to update the fields themselves to be `*.inodes` where appropriate, but we can't really do that without breaking existing mappings, and aliasing the fields seems confusing in its own right, as `files` and `inodes` are conceptually rather different.

## Why is it important?

This is a doc change that more accurately describes what some of the fields _actually_ mean.

## Checklist

- [x] I have made corresponding changes to the documentation
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Comment by cmacknz** [comment]
The other option here would be to have the metrics actually report the number of files instead of inodes, and add new metrics for inodes.

I don't think this affects correcting the documentation, but should we have an issue to consider doing?

File: metricbeat/docs/fields.asciidoc [hunk 1]
```diff
@@ -59864,7 +59864,7 @@ type: keyword
 *`system.filesystem.files`*::
 +
 --
-The total number of file nodes in the file system.
+Total number of inodes on the system, which will be a combination of files, folders, symlinks, and devices.
```
Comment (cmacknz) [inline metricbeat/docs/fields.asciidoc:59867]: Is this obvious in the code doing this as well? Should we add a comment to the code itself calling this out?

**Review by belimawr — APPROVED** [review]

**Comment by fearful-symmetry** [comment]
@cmacknz so, there's no good, OS-independent, filesystem-independent way to get the number of files. Doing something equivalent `find -type f` can be super slow and potentially hammer metricbeat and/or the filesystem. We could potentially derive something, but I'm not sure how much more accurate that would be than just telling users that the inode count is probably close.

**Comment by cmacknz** [comment]
> @cmacknz so, there's no good, OS-independent, filesystem-independent way to get the number of files. Doing something equivalent `find -type f` can be super slow and potentially hammer metricbeat and/or the filesystem. We could potentially derive something, but I'm not sure how much more accurate that would be than just telling users that the inode count is probably close.

Thanks, that makes sense. If you are updating the code comments can you add this explanation there as well? That will save the next person trying to understand this some time.

File: metricbeat/module/system/filesystem/helper.go [hunk 2]
```diff
@@ -111,6 +111,8 @@ func filterFileSystemList(fsList []sigar.FileSystem) []sigar.FileSystem {
 
 // GetFileSystemStat retreves stats for a single filesystem
 func GetFileSystemStat(fs sigar.FileSystem) (sigar.FileSystemUsage, error) {
+	// Sigar, in line with the underlying `statfs` call on unix, refers to inodes as `files` in the fields.
+	// There's no performant, filesystem-independent way to get *just* the count of files, so for most uses inodes will be close enough for jisting.
```
Comment (cmacknz) [inline metricbeat/module/system/filesystem/helper.go:115]:
```suggestion
	// There's no performant, filesystem-independent way to get *just* the count of files, so for most uses inodes will be close enough.
```

Is jisting a word? I don't know, but the sentence works fine without it.

Thanks for the comments!

Comment (fearful-symmetry) [reply metricbeat/module/system/filesystem/helper.go:115]: I was thinking of "gist" and somehow ended up there. I think I need more caffeine...

**Comment by v1v** [comment]
@fearful-symmetry ,  I'm afraid the backport-v8.2.0 is not yet available, and likely it points to main since main is 8.2 based. What branch would you need to backport this particular PR to? The backport definitions can be found in https://github.com/elastic/beats/blob/main/.mergify.yml#L210-L248

**Pull request merged**